connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockPlayerInputTranslator.java

```diff
@@ -44,5 +52,42 @@ public void translate(PlayerInputPacket packet, GeyserSession session) {
         );
 
         session.sendDownstreamPacket(clientSteerVehiclePacket);
+
+        // Bedrock only sends movement vehicle packets while moving
+        // This allows horses to take damage while standing on magma
+        Entity vehicle = session.getRidingVehicleEntity();
+        boolean sendMovement = false;
+        if (vehicle instanceof AbstractHorseEntity && !(vehicle instanceof LlamaEntity)) {
+            sendMovement = vehicle.isOnGround();
+        } else if (vehicle instanceof BoatEntity) {
+            if (vehicle.getPassengers().size() == 1) {
+                // The player is the only rider
+                sendMovement = true;
+            } else {
+                // Check if the player is the front rider
+                Vector3f seatPos = session.getPlayerEntity().getMetadata().getVector3f(EntityData.RIDER_SEAT_POSITION);
+                if (seatPos != null && seatPos.getX() > 0) {
+                    sendMovement = true;
+                }
+            }
+        }
+        if (sendMovement) {
+            long timeSinceVehicleMove = System.currentTimeMillis() - session.getLastVehicleMoveTimestamp();
+            if (timeSinceVehicleMove >= 100) {
+                Vector3f vehiclePosition = vehicle.getPosition();
+                Vector3f vehicleRotation = vehicle.getRotation();
+
+                if (vehicle instanceof BoatEntity) {
+                    // Remove some Y position to prevents boats flying up
+                    vehiclePosition = vehiclePosition.down(EntityType.BOAT.getOffset());
```

Comment:
This is inconsistent with the offset we currently apply to boats (https://github.com/GeyserMC/Geyser/blob/master/connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockMoveEntityAbsoluteTranslator.java#L48). I don't think the offset I chose was *perfect*, but the two should probably be consistent.

Reply:
Logging the two values seem to give the same answer.